### PR TITLE
topology2: cavs-passthrough-hdmi: Add support for chained DMA

### PR DESCRIPTION
**Pull request by ranj063**
Add support for chaind DMA pipelines in the HDMI passthrough topology. This version should work even if the kernel does not support chained DMA as yet.

Signed-off-by: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>

### CODE REVIEW

File: tools/topology/topology2/cavs/CMakeLists.txt [hunk 1]
```diff
@@ -3,23 +3,24 @@
 # Array of "input-file-name;output-file-name;comma separated pre-processor variables"
```
Comment (kv2019i) [inline tools/topology/topology2/cavs/CMakeLists.txt:3]: typo: chaind

**Review by kv2019i — APPROVED** [review]

**Review by lgirdwood — APPROVED** [review]
@jsarha pls review

**Review by fredoh9 — APPROVED** [review]

**Review by jsarha — APPROVED** [review]
I do not yet understand the details of topology configuration files, but this PR works with latest version on my chained DMA Linux PR: https://github.com/thesofproject/linux/pull/3953
I tested too that this version also works with current sof-dev head without my PR.

File: tools/topology/topology2/cavs/platform/intel/hdmi-4th.conf [hunk 2]
```diff
@@ -30,6 +31,7 @@ Object.Pipeline {
 			dai_type "HDA"
 			copier_type "HDA"
 		}
+		use_chain_dma	$USE_CHAIN_DMA
```
Comment (RanderWang) [inline tools/topology/topology2/cavs/platform/intel/hdmi-4th.conf:34]: @ranj063  if chain dma is enabled, do we need to disable S24_LE in pcm_caps ?

Comment (ranj063) [reply tools/topology/topology2/cavs/platform/intel/hdmi-4th.conf:34]: @RanderWang AFAIK there's no restriction for not having S24_LE with chained DMA. I know it doesnt work well today but we need to fix it

Comment (ranj063) [reply tools/topology/topology2/cavs/platform/intel/hdmi-4th.conf:34]: ok looks like we do have the restriction. I will remove S24_Le with chained DMA

**Review by plbossart — APPROVED** [review]

**Review by RanderWang — APPROVED** [review]

**Pull request merged**